app/models/smart_proxy.rb

```diff
@@ -112,20 +112,42 @@ def sanitize_url
   end
 
   def associate_features
-    return true if Rails.env == 'test'
-
     begin
-      reply = ProxyAPI::Features.new(:url => url).features
-      if reply.is_a?(Array) and reply.any?
-        self.features = Feature.where(:name => reply.map{|f| Feature.name_map[f]})
-      else
-        self.features.clear
-        errors.add :base, _('No features found on this proxy, please make sure you enable at least one feature')
-      end
+      proxy_response = ProxyAPI::Features.new(:url => url).features
     rescue => e
       errors.add(:base, _('Unable to communicate with the proxy: %s') % e)
       errors.add(:base, _('Please check the proxy is configured and running on the host.'))
     end
-    features.any?
+
+    return unless features_present?(proxy_response)
+    self.features = find_features(proxy_response)
+    unknown_features?(proxy_response, features)
+  end
+
+  def find_features(proxy_response)
+    Feature.where(:name => proxy_response.map{ |f| Feature.name_map[f] } )
+  end
+
+  def unknown_features?(proxy_response, features)
+    unknown_features = proxy_response - features.map(&:name).map(&:downcase)
+    if unknown_features.present?
+      errors.add(:base,
+                 _("None of the features found on this proxy"\
+                   "(%{unknown_features}) can be recognized by Foreman. "\
```

Comment:
Do we need to mention that it's not recognized by Foreman? Also, it doesn't have to be just becuse of the foreman version, but also just a plugin missing/disabled. I would suggest simplifying the error message to not cause some confusion. `"Features %{unknown_features} not recognized"`.